csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -159,6 +159,11 @@ class PredicateAnalyzer : public OptOutDispatch {
       return;
     }
 
+    if (factor.value() == 1) {
+      // Trivial splits cannot cause out-of-bounds
+      return;
+    }
+
```

Comment:
This is unrelated to this PR but I noticed the opportunity for this while debugging.